runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -688,53 +712,28 @@ private void checkClusterId(@Nonnull Layout layout) {
                     try {
                         IClientRouter router = getRouter(s);
                         // Try to get a layout.
-                        CompletableFuture<Layout> layoutFuture = router
-                                .getClient(LayoutClient.class).getLayout();
+                        CompletableFuture<Layout> layoutFuture =
+                                new LayoutSenderClient(router, Layout.INVALID_EPOCH).getLayout();
                         // Wait for layout
                         Layout l = layoutFuture.get();
 
-                        // If the layout we got has a smaller epoch than the router,
+                        // If the layout we got has a smaller epoch than the latestLayout epoch,
                         // we discard it.
-                        if (l.getEpoch() < router.getEpoch()) {
+                        if (latestLayout != null && latestLayout.getEpoch() > l.getEpoch()) {
```

Comment:
As we discussed we you can get rid of latestLayout.

Reply:
Removed.